src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+import { Env } from 'src/env/index';
+import { JwtPayload } from './jwt-interface';
+import { RedisService } from '@liaoliaots/nestjs-redis';
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy, 'jwt') {
+  constructor(
+    private readonly redisService: RedisService,
```

Comment:
PullProofAI

Consider adding error handling for Redis operations to handle potential failures.
```diff
+     try {
+         const client = this.redisService.getClient();
+         const jwtToken = await client.get(`jwt:${userId}`);
+         if (!jwtToken) {
+             throw new UnauthorizedException();
+         }
+     } catch (error) {
+         // Handle Redis operation errors
+         throw new UnauthorizedException('Failed to retrieve JWT token from Redis');
+     }
```



<!-- This is an auto-generated comment by PullProofAI -->